File: 250513/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] x = new int[n];
        int[] y = new int[n];
        for (int i = 0; i < n; i++) {
            x[i] = sc.nextInt();
            y[i] = sc.nextInt();
        }
        // Please write your code here.

        int min = Integer.MAX_VALUE;
        for (int i = 1; i < n-1; i++) {
            int minBucket = 0;
            int curX = x[0];
            int curY = y[0];
            for (int j = 0; j < n; j++) {
                if (j == i) continue;
                minBucket += Math.abs(x[j] - curX) + Math.abs(y[j] - curY);
                curX = x[j];
                curY = y[j];
            }
            min = Math.min(min, minBucket);
        }
        System.out.print(min);
    }
}

// 처음과 마지막 체크포인트는 건너뛰지 않을 것
// 하나 건너 뛰고 최소거리 구하기
// 체크포인트가 겹쳐져 있을 때는 그 번호의 체크포인트만 건너뛰는 것임
// 

File: 251022/마라톤 중간에 택시타기 2/taking-a-taxi-in-the-middle-of-the-marathon-2.py
n = int(input())
points = [tuple(map(int, input().split())) for _ in range(n)]
x = [p[0] for p in points]
y = [p[1] for p in points]

# Please write your code here.
import sys
minDistance = sys.maxsize
for i in range(1,n-1) :
    curX = x[0]
    curY = y[0]

    distance = 0

    for j in range(1,n) :
        if i == j :
            continue

        nx = x[j]
        ny = y[j]

        distance += abs(curX-nx) + abs(curY-ny)

        curX = nx
        curY = ny

    minDistance = min(minDistance, distance)
    
    
print(minDistance)

